modules/dnn/src/dnn_utils.cpp

```diff
@@ -42,90 +51,186 @@ void blobFromImages(InputArrayOfArrays images_, OutputArray blob_, double scalef
         Size size, const Scalar& mean_, bool swapRB, bool crop, int ddepth)
 {
     CV_TRACE_FUNCTION();
-    CV_CheckType(ddepth, ddepth == CV_32F || ddepth == CV_8U, "Blob depth should be CV_32F or CV_8U");
-    if (ddepth == CV_8U)
-    {
-        CV_CheckEQ(scalefactor, 1.0, "Scaling is not supported for CV_8U blob depth");
-        CV_Assert(mean_ == Scalar() && "Mean subtraction is not supported for CV_8U blob depth");
-    }
+    Image2BlobParams param(scalefactor, size, mean_, swapRB, ddepth);
+    if (crop)
+        param.paddingmode = DNN_PMODE_CROP_CENTER;
+    blobFromImagesWithParams(images_, blob_, param);
+}
+
+Mat blobFromImageWithParams(InputArray image, const Image2BlobParams& param)
+{
+    CV_TRACE_FUNCTION();
+    Mat blob;
+    blobFromImageWithParams(image, blob, param);
+    return blob;
+}
+
+void blobFromImageWithParams(InputArray image, OutputArray blob, const Image2BlobParams& param)
+{
+    CV_TRACE_FUNCTION();
+    std::vector<Mat> images(1, image.getMat());
+    blobFromImagesWithParams(images, blob, param);
+}
 
+Mat blobFromImagesWithParams(InputArrayOfArrays images, const Image2BlobParams& param)
+{
+    CV_TRACE_FUNCTION();
+    Mat blob;
+    blobFromImagesWithParams(images, blob, param);
+    return blob;
+}
+
+void blobFromImagesWithParams(InputArrayOfArrays images_, OutputArray blob_, const Image2BlobParams& param)
+{
+    CV_TRACE_FUNCTION();
+    CV_CheckType(param.ddepth, param.ddepth == CV_32F || param.ddepth == CV_8U,
+                 "Blob depth should be CV_32F or CV_8U");
+
+    Size size = param.size;
     std::vector<Mat> images;
     images_.getMatVector(images);
     CV_Assert(!images.empty());
+
+    int nch = images[0].channels();
+    Scalar scalefactor = param.scalefactor;
+
+    // Because 0 is meaningless in scalefactor.
+    // If the scalefactor is (x, 0, 0, 0), we convert it to (x, x, x, x).
+    if (scalefactor[1] == 0 && scalefactor[2] == 0 && scalefactor[3] == 0)
+    {
+        CV_Assert(scalefactor[0] != 0 && "Scalefactor of 0 is meaningless.");
+        scalefactor = Scalar::all(scalefactor[0]);
+    }
```

Comment:
The function was renamed from `blobFromImages` to `blobFromImagesWithParams`. No need to presume backward compatibility. I propose to drop the hack.

Reply:
Hi @vpisarev and @asmorkalov, what's the final decision of this hack?

Reply:
ok to drop it